lib/repl.js

```diff
@@ -1285,63 +1266,63 @@ function complete(line, callback) {
-          } catch {
-            // Probably a Proxy object without `getOwnPropertyNames` trap.
-            // We simply ignore it here, as we don't want to break the
-            // autocompletion. Fixes the bug
-            // https://github.com/nodejs/node/issues/2119
-          }
+      try {
+        let p;
+        if ((typeof obj === 'object' && obj !== null) ||
+            typeof obj === 'function') {
+          memberGroups.push(filteredOwnPropertyNames(obj));
+          p = ObjectGetPrototypeOf(obj);
+        } else {
+          p = obj.constructor ? obj.constructor.prototype : null;
         }
-        // Works for non-objects
-        try {
-          let p;
-          if (typeof obj === 'object' || typeof obj === 'function') {
-            p = ObjectGetPrototypeOf(obj);
-          } else {
-            p = obj.constructor ? obj.constructor.prototype : null;
-          }
-          // Circular refs possible? Let's guard against that.
-          let sentinel = 5;
-          while (p !== null && sentinel-- !== 0) {
-            memberGroups.push(filteredOwnPropertyNames(p));
-            p = ObjectGetPrototypeOf(p);
-          }
-        } catch {}
+        // Circular refs possible? Let's guard against that.
+        let sentinel = 5;
+        while (p !== null && sentinel-- !== 0) {
+          memberGroups.push(filteredOwnPropertyNames(p));
+          p = ObjectGetPrototypeOf(p);
+        }
+      } catch {
+        // Maybe a Proxy object without `getOwnPropertyNames` trap.
+        // We simply ignore it here, as we don't want to break the
+        // autocompletion. Fixes the bug
+        // https://github.com/nodejs/node/issues/2119
       }
 
       if (memberGroups.length) {
-        for (let i = 0; i < memberGroups.length; i++) {
-          completionGroups.push(
-            memberGroups[i].map((member) => `${expr}.${member}`));
+        expr += chaining;
+        for (const group of memberGroups) {
+          completionGroups.push(group.map((member) => `${expr}${member}`));
         }
         if (filter) {
-          filter = `${expr}.${filter}`;
+          filter = `${expr}${filter}`;
         }
       }
 
       completionGroupsLoaded();
     });
-  } else {
-    completionGroupsLoaded();
+    return;
   }
 
+  return completionGroupsLoaded();
+
   // Will be called when all completionGroups are in place
   // Useful for async autocompletion
   function completionGroupsLoaded() {
     // Filter, sort (within each group), uniq and merge the completion groups.
     if (completionGroups.length && filter) {
       const newCompletionGroups = [];
-      for (let i = 0; i < completionGroups.length; i++) {
-        group = completionGroups[i]
-          .filter((elem) => elem.indexOf(filter) === 0);
-        if (group.length) {
-          newCompletionGroups.push(group);
+      for (const group of completionGroups) {
+        const filteredGroup = group.filter((e) => e.indexOf(filter) === 0);
```

Comment:
`e` isn't really a great name here (elem wasn't either to be fair) :] If filter is a string an `e` is a string this can also be a startsWith

Reply:
I updated it to `str` instead and refactored more code for readability (e.g., using `startsWith` and `endsWith`).